src/main/java/com/celements/metatag/MetaTag.java

```diff
@@ -0,0 +1,45 @@
+package com.celements.metatag;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import org.xwiki.component.annotation.Component;
+import org.xwiki.component.annotation.Requirement;
+
+import com.celements.model.context.ModelContext;
+
+@Component
+public class MetaTag implements IMetaTag {
+
+  @Requirement
+  ModelContext modelContext;
+
+  @Override
+  public void addMetaTag(MetaTagApi tag) {
+    Object contextObj = modelContext.getXWikiContext().get(IMetaTag.META_CONTEXT_KEY);
+    if (contextObj instanceof List) {
+      @SuppressWarnings("unchecked")
+      List<MetaTagApi> metaList = (List<MetaTagApi>) contextObj;
+      metaList.add(tag);
+    } else {
+      List<MetaTagApi> metaList = new ArrayList<>();
+      metaList.add(tag);
+      modelContext.getXWikiContext().put(IMetaTag.META_CONTEXT_KEY, metaList);
```

Comment:
Better use Execution context instead of deprecated XWikiContext to store/cache the metaList for one request